projects/batfish/test/org/batfish/main/BatfishTest.java

```diff
@@ -0,0 +1,91 @@
+package org.batfish.main;
+
+import org.batfish.common.BatfishException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.collection.IsEmptyCollection.empty;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Test for {@link Batfish#listAllFiles(Path)}.
+ */
+public class BatfishTest {
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void throwsExceptionWithSpecificType() {
+        thrown.expect(BatfishException.class);
+        thrown.expectMessage("Failed to walk path: ./test/notExist");
```

Comment:
rather than making up a string, use something resolved from folder that you know does not exist. E.g., `folder.getRoot().toPath().resolve("nonexistent")`.